src/worker-vless.js

```diff
@@ -436,39 +466,55 @@ async function isCloudFlareIP(addressType, addressRemote) {
 		return false;
 	}
 
-	if (addressType === 1) {
-		return isIPv4InCFCIDR(addressRemote);
-	}
-
-  if(addressType === 2)
-  {
-    const domainIP = await doh(addressRemote);
-    console.log(' domainIP ', domainIP);
-    return isIPv4InCFCIDR(domainIP);
-  }
-  return false;
+    if(addressType === 2)
+    {
+        return await isBehindCFv6(addressRemote);
+    }
+    return false;
 }
 
 
 /**
  * 
  * @param {string} domain 
- * @returns 
+ * @returns {Promise<boolean>}
  */
-async function doh(domain) {
-	const doh = "https://cloudflare-dns.com/dns-query";
-	const response = await fetch(`${doh}?name=${domain}`, {
-		method: "GET",
-		headers: {
-			"Accept": "application/dns-json"
-		}
-	});
-	const data = await response.json();
-	const ans = data?.Answer;
-  //https://developers.cloudflare.com/1.1.1.1/encryption/dns-over-https/make-api-requests/dns-json/
-	return ans?.find((record) => record.type === 1)?.data;
+async function isBehindCFv6(domain) {
+	const doh = "https://1.1.1.1/dns-query";
+	try {
+		const response = await fetch(`${doh}?name=${domain}.cdn.cloudflare.net&type=AAAA`, {
+			method: "GET",
+			headers: {
+				"Accept": "application/dns-json"
+			}
+		});
+		const data = await response.json();
+		const ans = data?.Answer;
+		//https://developers.cloudflare.com/1.1.1.1/encryption/dns-over-https/make-api-requests/dns-json/
```

Comment:
Do we need put  domainByoListCheck(domain, byoList) before doh fetch?

Reply:
Sure thing. And tp improve accuracy we can replace the line 494 with: ```		return ans?.filter((record) => record.name === `${domain}.cdn.cloudflare.net` && record.type === 28).length > 1 || domainByoListCheck(domain, byoList); ```

Reply:
I mean, if we move `domainByoListCheck(domain, byoList)` before doh fetch, we don't need call doh for byo ip domain?

Reply:
Both ways would work but actually byoCheck won't hit in the most of case.